File: app/backend/src/Model/AgentProfile.php
<?php

namespace Nexus\Forum\Model;

use Flarum\Database\AbstractModel;
use Flarum\User\User;

class AgentProfile extends AbstractModel
{
    protected $table = 'nexus_agent_profiles';

    protected $primaryKey = 'user_id';

    public $incrementing = false;

    public $timestamps = true;

    protected $casts = [
        'user_id' => 'int',
        'allow_agent_posting' => 'bool',
        'allow_agent_replying' => 'bool',
        'allow_agent_matching' => 'bool',
        'allow_location_matching' => 'bool',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id');
    }
}

File: server/src/main/java/nexus/campus/agent/entity/AgentProfile.java
package nexus.campus.agent.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import nexus.campus.common.entity.User;

import java.time.LocalDateTime;

@Entity
@Table(name = "nexus_agent_profiles")
@Getter @Setter @NoArgsConstructor
public class AgentProfile {
    @Id
    @Column(name = "user_id")
    private Integer userId;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @Column(name = "agent_name", length = 120)
    private String agentName;

    @Column(name = "agent_avatar_url", length = 512)
    private String agentAvatarUrl;

    @Column(name = "soul_md", columnDefinition = "text")
    private String soulMd;

    @Column(name = "interest_tags", columnDefinition = "text")
    private String interestTags;

    @Column(name = "skill_tags", columnDefinition = "text")
    private String skillTags;

    @Column(name = "help_tags", columnDefinition = "text")
    private String helpTags;

    @Column(name = "match_preferences", columnDefinition = "text")
    private String matchPreferences;

    @Column(name = "allow_agent_posting")
    private boolean allowAgentPosting;

    @Column(name = "allow_agent_replying")
    private boolean allowAgentReplying;

    @Column(name = "allow_agent_matching")
    private boolean allowAgentMatching;

    @Column(name = "allow_location_matching")
    private boolean allowLocationMatching;

    @Column(name = "location_visibility", length = 40)
    private String locationVisibility = "off";

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() { createdAt = LocalDateTime.now(); updatedAt = LocalDateTime.now(); }

    @PreUpdate
    protected void onUpdate() { updatedAt = LocalDateTime.now(); }
}
